### PR TITLE
Reuse error channel

### LINKED ISSUE #2431: [SLES15]: Fleet-server Agent gets into offline state on machine reboot.
**Kibana version:** 8.7.0 Prod Kibana cloud environment

**Host OS:** SLES

**Build details:**
VERSION: 8.7.0 BC5
BUILD: 61017
COMMIT: e70452f9697a18a8458f3d4618e001b7e3681bed
Artifact: https://staging.elastic.co/8.7.0-69b36bf6/summary-8.7.0.html

**Preconditions:**

- 8.7.0 Prod Kibana cloud environment should be available.
- Agent Policy should be available with System and Fleet-server integrations.

**Steps to reproduce:**

- Install agent using above mentioned agent policy.
- Wait for agent to generate logs and data streams properly on Fleet UI.
- Reboot the machine after 5 minutes.
- Observe Fleet-server Agent gets into offline state and never comes back to Healthy status.

**Screenshots:**
image
- Though agent is seen as running on Host VM
image

**Testcase Reference:**
https://elastic.testrail.io/index.php?/tests/view/2056682

**Logs:**
[elastic-agent-diagnostics-2023-03-15T08-11-08Z-00.zip](https://github.com/elastic/elastic-agent/files/10977493/elastic-agent-diagnostics-2023-03-15T08-11-08Z-00.zip)

### PR DESCRIPTION
**Pull request by michel-laterman**
## What is the problem this PR solves?

SLES instances running a fleet-server instance under the elastic-agent are unhealthy after a restart (fleet-server does not reconfigure properly)

## How does this PR solve the problem?

The error channel  used by the server struct is being created on each iteration of the run loop: https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L136

However, [runServer](https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L183) call actually occurs in a different goroutine: https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L117.

Which can lead to a [config change being read](https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L190) before the runServer method is ran and leaking the channel; the error passed to the channel is lost and thus the fleet-server is stuck as it should signal that it's failed but has not.

## How to test this PR locally

Install an agent with the fleet-server integration on a SLES machine and restart the machine.

Without the changes the logs should stop with:
```
{"log.level":"info","@timestamp":"2023-06-12T22:16:42.131Z","log.origin":{"file.name":"coordinator/coordinator.go","file.line":985},"message":"Unit state changed fleet-server-default-fleet-server-fleet_server-9baba0cd-d2de-47cc-8a9b-d1d2f8c12d66 (STARTING->CONFIGURING): Re-configuring","log":{"source":"elastic-agent"},"component":{"id":"fleet-server-default","state":"HEALTHY"},"unit":{"id":"fleet-server-default-fleet-server-fleet_server-9baba0cd-d2de-47cc-8a9b-d1d2f8c12d66","type":"input","state":"CONFIGURING","old_state":"STARTING"},"ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2023-06-12T22:16:42.131Z","log.origin":{"file.name":"coordinator/coordinator.go","file.line":985},"message":"Unit state changed fleet-server-default (STARTING->CONFIGURING): Re-configuring","log":{"source":"elastic-agent"},"component":{"id":"fleet-server-default","state":"HEALTHY"},"unit":{"id":"fleet-server-default","type":"output","state":"CONFIGURING","old_state":"STARTING"},"ecs.version":"1.6.0"}

# This error is lost
{"log.level":"error","@timestamp":"2023-06-12T22:16:42.132Z","message":"fail elasticsearch info","component":{"binary":"fleet-server","dataset":"elastic_agent.fleet_server","id":"fleet-server-default","type":"fleet-server"},"log":{"source":"fleet-server-default"},"cluster.maxConnsPersHost":128,"error.message":"dial tcp: lookup 200dd16c33174ccb94de816bb6fd5c18.us-central1.gcp.qa.cld.elstc.co on [::1]:53: read udp [::1]:38166->[::1]:53: read: connection refused","ecs.version":"1.6.0","service.name":"fleet-server","cluster.addr":["200dd16c33174ccb94de816bb6fd5c18.us-central1.gcp.qa.cld.elstc.co:443"],"ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2023-06-12T22:16:42.132Z","message":"Stats endpoint (/opt/Elastic/Agent/data/tmp/fleet-server-default.sock) finished: accept unix /opt/Elastic/Agent/data/tmp/fleet-server-default.sock: use of closed network connection","component":{"binary":"fleet-server","dataset":"elastic_agent.fleet_server","id":"fleet-server-default","type":"fleet-server"},"log":{"source":"fleet-server-default"},"ecs.version":"1.6.0","service.name":"fleet-server","log.logger":"fleet-metrics.api","ecs.version":"1.6.0"}
```

With the changes the logs will have the `Fleet Server failed` message after.

## Checklist

- ~~I have commented my code, particularly in hard-to-understand areas~~
- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~
- ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

## Related issues

- Closes #2431

### CODE REVIEW

**Review by jsoriano — COMMENTED** [review]
Would it be possible to add a test case for this?

**Comment by juliaElastic** [comment]
Good catch! The fix looks good, we should add an integration test if possible to simulate the restart scenario.

File: changelog/fragments/1686780651-reuse-error-channel.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#description:
+
+# Affected component; a word indicating the component this changeset affects.
+component:
+
+# PR URL; optional; the PR number that added the changeset.
+# If not present is automatically filled by the tooling finding the PR where this changelog fragment has been added.
+# NOTE: the tooling supports backports, so it's able to fill the original PR number instead of the backport PR number.
+# Please provide it if you are adding a fragment for a different PR.
+#pr: https://github.com/owner/repo/1234
```
Comment (juliaElastic) [inline changelog/fragments/1686780651-reuse-error-channel.yaml:28]:
```suggestion
#pr: 2697
```

File: internal/pkg/server/fleet_integration_test.go [hunk 2]
```diff
@@ -226,6 +252,123 @@ func (s *tserver) buildURL(id string, cmd string) string {
...
+	mReporter.On("UpdateState", client.UnitStateStopping, mock.Anything, mock.Anything).Return(nil)
+
+	// set bad config
+	cfg.Output.Elasticsearch.ServiceToken = "incorrect"
+
+	// send good config
+	err = srv.Reload(ctx, newCfg)
+	require.NoError(t, err)
+
+	// Run server with bad config - it should fail, then work on the restart?
```
Comment (juliaElastic) [inline internal/pkg/server/fleet_integration_test.go:357]: can we remove commented out code?

**Pull request merged**